### PR TITLE
docstring update to lexicographical_topological_sort issue 5681

### LINKED ISSUE #5681: treewidth algos depend on node types
The treewidth functions appear to depend on the types of nodes in the input graph.

### Current Behavior

For example, the algos fail is there is a mix of nodes with integer and string types.

### Expected Behavior

The algos should not depend on node types, only the graph structure.

### Steps to Reproduce

```python
import networkx as nx
from networkx.algorithms.approximation import treewidth
G = nx.Graph()
G.add_nodes_from([0, 'a'])
tw, td = treewidth.treewidth_min_degree(G)
or
tw, td = treewidth.treewidth_min_fill_in(G)
```

### Environment

Python version: 3.8
NetworkX version:  2.8

### Additional context

Error:
  File "<somedir>/lib/python3.8/site-packages/networkx/algorithms/approximation/treewidth.py", line 106, in __init__
    heapify(self._degreeq)
TypeError: '<' not supported between instances of 'int' and 'str'

### PR DESCRIPTION
**Pull request by brocla**
Added docstring improvements to lexicographical_topological_sort as discussed in issue 5681. Partially resolves #5681 

- explains effect and purpose for lexi sort in the short and long descriptions.
- add hints for fixing un-sortable nodes
- adds hint to exception msg
- Adds example to demonstrate fixing un-sortable nodes with key function

### CODE REVIEW

File: networkx/algorithms/dag.py [hunk 1]
```diff
@@ -347,6 +364,26 @@ def lexicographical_topological_sort(G, key=None):
     >>> list(nx.lexicographical_topological_sort(DG, key=lambda x: -x))
     [2, 5, 1, 4, 3]
 
+    The sort will fail for this graph because the comparison of integers to strings
```
Comment (dschult) [inline networkx/algorithms/dag.py:367]:
```suggestion
    The sort will fail for any graph with integer and string nodes. Comparison of integer to strings
```

File: networkx/algorithms/dag.py [hunk 2]
```diff
@@ -304,26 +304,39 @@ def topological_sort(G):
...
     Parameters
     ----------
     G : NetworkX digraph
         A directed acyclic graph (DAG)
 
     key : function, optional
-        This function maps nodes to keys with which to resolve ambiguities in
-        the sort order.  Defaults to the identity function.
+        A function of one argument that converts a node name to a comparison key.
+        Use to resolve ambiguities in the sort order.  Defaults to the identity function.
```
Comment (dschult) [inline networkx/algorithms/dag.py:334]:
```suggestion
        It defines and resolves ambiguities in the sort order.  Defaults to the identity function.
```

File: networkx/algorithms/dag.py [hunk 3]
```diff
@@ -347,6 +364,26 @@ def lexicographical_topological_sort(G, key=None):
...
+    >>> DG = nx.DiGraph([(1, 'red'), (3, 'red'), (1, 'green'), (2, 'blue')])
+    >>> list(nx.lexicographical_topological_sort(DG))
+    Traceback (most recent call last):
+    ...
+    TypeError: '<' not supported between instances of 'str' and 'int'
+    ...
+
+    The solution is to provide a function that returns keys that do compare.
+
+    There are many ways to write a `key` function. This one returns a tuple where the first
```
Comment (dschult) [inline networkx/algorithms/dag.py:379]:
```suggestion
    Incomparable nodes can be resolved using a `key` function. This example function 
    allows comparison of integers and strings by returning a tuple where the first
```

**Review by dschult — COMMENTED** [review]
This looks good--Thanks!
I've done some word-smithing below. Push back on anything that you don't view as an improvement and feel free to make improvements upon these suggestions.

**Comment by brocla** [comment]
Hmmm... A check failed.   It says that Black failed. Not sure why.

**Comment by dschult** [comment]
It looks one of my suggestions had white space at the end of two lines.
So black is removing it.  You can run black locally on `dag.py` and it should fix the program. If you feel more comfortable knowing what it will change before it changes it, you can run 
`black --diff dag.py` and it shows you the changes it will make.

To automatically run black and isort (our import sorting style checker) before every commit you make, you can use pre-commit.  To install on a `pip` installing environment:
```
pip install pre-commit
```
To install in a conda environment:
```
conda install -c conda-forge pre-commit
```
Then once it is installed, within your environment, install it in the local repository you are working on.  cd to somewhere in the repo, then ```pre-commit install```.

If that is too much trouble and too automatic, you can run `black` locally, installing with pip or conda and then  `black --diff dag.py` followed by `black dag.py`.

**Review by dschult — APPROVED** [review]
This looks good to me! 
Thanks!

**Comment by brocla** [comment]
The tests are passing now. Good.

I think I missed the benefits of pre-commit because I accepted the changes on github. 

Anyway, I pulled the changes back to my local machine. Ran Black. Committed. And pushed to upstream.  The tests suite re-ran. All passing. :-)

Thanks again for the review.

**Review by MridulS — APPROVED** [review]
This is great! Thanks @brocla

**Pull request merged**